util/src/main/java/com/google/cloud/hadoop/util/HttpTransportFactory.java

```diff
@@ -138,6 +145,9 @@ public static ApacheHttpTransport createApacheHttpTransport(
 
     HttpClientBuilder httpClientBuilder = ApacheHttpTransport.newDefaultHttpClientBuilder();
 
+    SocketConfig keepAliveSocketConf = SocketConfig.custom().setSoKeepAlive(true).build();
+    httpClientBuilder.setDefaultSocketConfig(keepAliveSocketConf);
```

Comment:
let's add a test to cover this path

Reply:
I tried to find a way to test this but couldn't. So this config goes into building an [HttpClient](https://javadoc.io/doc/org.apache.httpcomponents/httpclient/latest/org/apache/http/client/HttpClient.html) which doesn't expose any method to expose this config.

I think it is fine to not unit test this config as it is obvious and we anyway don't have this in future versions. Also, for this version it is not default.

Reply:
wdyt?

Reply:
ack